bin/cylc-check-software

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
```

Comment:
Out of curiosity, why did we bring just the `cylc-check-software` with this PR?

Reply:
On this branch, I stripped everything out of `cylc-check-software` except for the doc build requirements.  Not necessarily thinking we should keep it here, I was just trying to do a minimal split of our existing functionality as a first cut.
```
[oliverh@niwa-1007823l cylc-doc]$ bin/cylc-check-software 
Checking your software...

Individual results:
==========================================================================================
Package (version requirements)                                     Outcome (version found)
==========================================================================================
                                   *REQUIRED SOFTWARE*                                   
Python (3.7+).....................................FOUND & min. version MET (3.7.3.final.0)

                      *OPTIONAL SOFTWARE for the HTML documentation*                      
Python:sphinx (1.5.3+).......................................................NOT FOUND (-)
==========================================================================================

Summary:
                               ****************************                               
                                  Core requirements: ok                                  
                                Full-functionality: not ok                                
                               ****************************                               
```

Reply:
(And in the cylc-flow PR, I stripped the doc build bits out of `cylc-check-software`)

Reply:
:+1: got it! Makes sense now, thanks!